x/bor/keeper/msg_server.go

```diff
@@ -259,6 +259,39 @@ func (s msgServer) BackfillSpans(ctx context.Context, msg *types.MsgBackfillSpan
 	return &types.MsgBackfillSpansResponse{}, nil
 }
 
+func (s msgServer) SetProducerDowntime(ctx context.Context, msg *types.MsgSetProducerDowntime) (*types.MsgSetProducerDowntimeResponse, error) {
+	var err error
+	start := time.Now()
+	defer recordBorTransactionMetric(api.ProducerDowntimeMethod, start, &err)
+
+	validators := s.sk.GetSpanEligibleValidators(ctx)
```

Comment:
If the proposer is not part of eligible producer, we should reject immediately. This will avoid unnecessary computation in case a validator who is outside of producer set spamming the network with such proposals.